drivers/flash/spi_nor.c

```diff
@@ -702,6 +702,34 @@ static int spi_nor_read(const struct device *dev, off_t addr, void *dest,
 	return ret;
 }
 
+#if defined(CONFIG_FLASH_EX_OP_ENABLED)
+static int flash_spi_nor_ex_op(const struct device *dev, uint16_t code,
+			const uintptr_t in, void *out)
+{
+	int ret;
+
+	ARG_UNUSED(in);
+	ARG_UNUSED(out);
+
+	acquire_device(dev);
+
+	switch (code) {
+	case FLASH_SPI_NOR_RESET:
+		ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_RESET_EN);
+		if (ret == 0) {
+			ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_RESET_MEM);
+		}
+		break;
+	default:
+		ret = -ENOTSUP;
+		break;
+	}
```

Comment:
I think you are missing here the CONFIG_USERSPACE support. Look how is the other ex_op implementation done, for example for the stm32 devices rd protections.

Reply:
When i enable CONFIG_FLASH_EX_OP_ENABLED and verify it, its functional as expected. Flash reset happens.
Can you please provide more details on what is missing here?

Reply:
enabled the FLASH_HAS_EX_OP in file drivers/flash/Kconfig.nor